src/EStimLibrary/Core/ReusableIdPool.cs

```diff
@@ -102,13 +122,19 @@ public ReusableIdPool()
 
     public int IncrementNumIds(int increment)
     {
-        this.NumIds += increment;
-        return this.NumIds;
+        if (this.NumIds + increment <= int.MaxValue)
```

Comment:
BUG: should not allow "numIds" to go below the current highest Used ID. This can be another part of the NumIds set() method

Reply:
Once have updated the NumIds set() method with both adjustments (min and max bound checking), can remove this if-else and simply edit NumIds and then return its value from the method

Reply:
Once both of these changes are made, add test cases for them

Reply:
When you say NumIds set() method, do you mean this:
/// <summary>
/// The number of IDs in this pool. Cannot be below 0.
/// </summary>
public int NumIds
{
    get => this._numIds;
    protected set => this._numIds = Math.Max(MIN_NUM_IDS, value);
}

If so, does that mean that I can change the "set" functionality inside?

Reply:
I have pushed a commit fixing this. Please check if my modification is good